swig/include/python/osr_python.i

```diff
@@ -53,9 +53,34 @@ def _WarnIfUserHasNotSpecifiedIfUsingExceptions():
 
         _WarnIfUserHasNotSpecifiedIfUsingExceptions()
 
+        user_input = None
+
+        if len(args) + len(kwargs) > 1:
+            # We could pass additional kwargs as options to SetFromUserInput,
+            # but that is not commonly needed and limits our ability to
+            # validate arguments here.
+            raise ValueError("Unexpected argument to SpatialReference")
+
+        if kwargs:
+            if "wkt" in kwargs:
+                user_input = kwargs["wkt"]
+            elif "name" in kwargs:
+                user_input = kwargs["name"]
+            elif "epsg" in kwargs:
+                user_input = f'EPSG:{kwargs["epsg"]}'
+            else:
+                raise ValueError("Unexpected argument to SpatialReference")
+
+        if args:
+            if type(args[0]) is dict:
+                import json
+                user_input = json.dumps(args[0])
+            else:
+                user_input = args[0]
+
         try:
             with ExceptionMgr(useExceptions=True):
-                this = _osr.new_SpatialReference(*args, **kwargs)
+                this = _osr.new_SpatialReference()
```

Comment:
the try / finally / ExceptionMgr is likely unnecessary now that we no longer pass arguments to the constructor.  And probably all the below stuff until line 96

Reply:
> And probably all the below stuff until line 96

actually I'm probably wrong about that. That must be standard SWIG initialization